src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -96,7 +95,9 @@ List<LogicalMessage> Extract(IncomingMessage physicalMessage)
             using (var stream = new MemoryStream(physicalMessage.Body))
             {
                 var messageTypes = messageMetadata.Select(metadata => metadata.MessageType).ToList();
-                var messageSerializer = deserializerResolver.Resolve(physicalMessage.Headers[Headers.ContentType]);
+                string contentType;
+                physicalMessage.Headers.TryGetValue(Headers.ContentType, out contentType);
+                var messageSerializer = deserializerResolver.Resolve(contentType);
```

Comment:
Should we throw a better exception here in cases where content type is not known and we attempt the default one and that still fails ?